### PR TITLE
Resource: service

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/67

## Changes
- added service-account resource

### CODE REVIEW

File: twingate/internal/provider/resource/service-account.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+	"github.com/Twingate/terraform-provider-twingate/twingate/internal/client"
+	"github.com/Twingate/terraform-provider-twingate/twingate/internal/model"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Services offer a way to provide programmatic, centrally-controlled, and consistent access controls. For more information, see Twingate's [documentation](https://www.twingate.com/docs/services).",
```
Comment (vmanilo) [inline twingate/internal/provider/resource/service-account.go:16]: @alexmensch please check description

File: docs/resources/service_account.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_service_account Resource - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  Services offer a way to provide programmatic, centrally-controlled, and consistent access controls. For more information, see Twingate's documentation https://www.twingate.com/docs/services.
```
Comment (alexmensch) [inline docs/resources/service_account.md:6]: I know this is taken from the public docs, but let's do this everywhere: s/Services/Service Accounts/g

In this description, let's remove the sentence "For more information [...]"

File: docs/resources/service_account.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+# twingate_service_account (Resource)
+
+Services offer a way to provide programmatic, centrally-controlled, and consistent access controls. For more information, see Twingate's [documentation](https://www.twingate.com/docs/services).
```
Comment (alexmensch) [inline docs/resources/service_account.md:11]: s/Services/Service Accounts/

File: docs/resources/service_account.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  name = "Github Actions PROD"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of the service account in Twingate
```
Comment (alexmensch) [inline docs/resources/service_account.md:31]: s/service account/Service Account/

File: docs/resources/service_account.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+### Read-Only
+
+- `id` (String) Autogenerated ID of the service account
```
Comment (alexmensch) [inline docs/resources/service_account.md:35]: s/service account/Service Account/

**Review by alexmensch — CHANGES_REQUESTED** [review]
Some simple changes (I'll make the same changes in the public documentation)

File: docs/resources/service_account.md (hunk 2 above)
Comment (vmanilo) [reply docs/resources/service_account.md]: @alexmensch I've updated doc strings

File: docs/resources/service_account.md (hunk 3 above)
Comment (vmanilo) [reply docs/resources/service_account.md]: updated

File: docs/resources/service_account.md (hunk 4 above)
Comment (vmanilo) [reply docs/resources/service_account.md]: updated

File: docs/resources/service_account.md (hunk 5 above)
Comment (vmanilo) [reply docs/resources/service_account.md]: updated

File: twingate/internal/client/pagination_test.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
+package client
```
Comment (ekampf) [inline twingate/internal/client/pagination_test.go:1]: Why is this here and not in `twingate/internal/test/client` ?

File: twingate/internal/client/converter.go [hunk 7]
```diff
@@ -230,3 +230,31 @@ func (r ResourceNode) ToModel() *model.Resource {
 func (q readResourcesByNameQuery) ToModel() []*model.Resource {
 	return q.Resources.ToModel()
 }
+
```
Comment (ekampf) [inline twingate/internal/client/converter.go:233]: Its something I missed in the restructure review but Im not sure I understand the purpose of `converter.go` - we define `createServiceAccountQuery` in `service-account.go` why cant its methods be there?  
(same for all the other structs that their methods are here)

File: twingate/internal/client/pagination_test.go (hunk 6 above)
Comment (vmanilo) [reply twingate/internal/client/pagination_test.go:1]: here I'm testing private func `fetchPages` and it's not accessible from other packages

File: twingate/internal/client/converter.go (hunk 7 above)
Comment (vmanilo) [reply twingate/internal/client/converter.go:233]: here in `converter.go` I've collected simple utilities functions which converts queries objects to models

I think in `service-account.go` would be better to have only code related to API request, and this object conversions like lower layer details

I like the idea with separate package for `queries`, I think this can help organise code better :)

Comment (ekampf) [reply twingate/internal/client/converter.go:233]: ok so lets merge this and refactor to `queries` on a separate PR
